File: Accommodation-Reservation-Backend/src/main/java/com/accommodationsite/accommodationreservationapp/model/Person.java
package com.accommodationsite.accommodationreservationapp.model;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Person implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    @Column(nullable = false)
    private String username;
    @Column(nullable = false)
    private String password;
    private String firstName;
    private String lastName;
    @Column(nullable = false)
    private String email;
    private String phone;
    private String role;
    private Boolean activated;
    @OneToOne(targetEntity = Address.class, cascade = CascadeType.MERGE)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    private Address address;
}

File: Accommodation-Reservation-Frontend/src/app/shared/models/person.ts
import { Address } from "./address";

export interface Person {
    id?: number;
    username: string;
    password: string;
    firstName?: string;
    lastName?: string;
    email: string;
    phone?: string;
    role?: string;
    activated?: boolean;
    own_accommodation_id?: number;
    address?: Address;
}